docs/developer-guide/views-examples.md

```diff
@@ -0,0 +1,120 @@
+---
+description: Learn about the most used custom view examples of Lite XL.
+---
+
+## Toolbar
+
+### Tutorial
+
+```lua
+-- mod-version:3 -- lite-xl 2.1
+```
+
+This is required to specify Lite XL version compatibility.
+
+The instructions that begin with `require` add necessary libraries.
+
+```lua
+local toolbar = common.merge({
+  -- plugin configuration options go here
+}, config.plugins.build)
+```
+
+The code above contains the (optional) configuration options of the plugin.
+
+The `get_plugin_directory()` function gets the plugin folder path and tells the 
+plugin where to find the icon font.
+The iteration code is used to check if every folder detected by `get_plugin_directory()` 
+actually exists and if the folder doesn't exist, a `nil` value is returned.
+
+`local ToolBar = ToolbarView:extend()` is used to create a new custom View and make it 
+inherit ToolbarView.
+
+```lua
+function ToolBar:new()
+  ToolBar.super.new(self)
+  self.toolbar_font = renderer.font.load(get_plugin_directory() .. PATHSEP .. "toolbar.ttf", style.icon_big_font:get_size())
+  self.toolbar_commands = {
+    {symbol = "", command = "core:open-log"},
+    {symbol = "", command = "core:new-doc"},
+    {symbol = "", command = ""},
+    {symbol = "", command = ""},
+    {symbol = "", command = ""},
+    {symbol = "", command = ""},
+  }
+end
+```
+
+The code above creates an instance of our custom Toolbar view, inherits the ToolbarView super-constructor, 
+specifies the desired icon font and assigns the icons to their respective commands.
+
+`toolbar.example_toolbar_view = ToolBar()` saves the toolbar in an exported table, to allow any plugin to require it, thus accessing the bar.
+
+```lua
+toolbar.example_toolbar_node = TreeView.node.b:split("up", toolbar.example_toolbar_view, {y = true})
+```
+
+This instruction splits the `TreeView` according to the `up` direction, adds the `toolbar.example_toolbar_view` and fixes its size along the y axis.
+
+`return toolbar` returns the toolbar table to Lite XL.
```

Comment:
no need to explain this, also it doesn't return to Lite XL, it just returns to whatever the caller happens to be.